contracts/DEXes/UniV3StablesJoint.sol

```diff
@@ -0,0 +1,639 @@
+                0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000
+            )
+            mstore(add(clone_code, 0x14), addressBytes)
+            mstore(
+                add(clone_code, 0x28),
+                0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000
+            )
+            newJoint := create(0, clone_code, 0x37)
+        }
+
+        UniV3StablesJoint(newJoint).initialize(
+            _providerA,
+            _providerB,
+            _referenceToken,
+            _pool,
+            _ticksFromCurrent
+        );
+
+        emit Cloned(newJoint);
+    }
+
+    /*
+     * @notice
+     *  Function returning the name of the joint in the format "NoHedgeUniV3StablesJoint(USDC-DAI)"
+     * @return name of the strategy
+     */
+    function name() external view override returns (string memory) {
+        string memory ab = string(
+            abi.encodePacked(
+                IERC20Extended(address(tokenA)).symbol(),
+                "-",
+                IERC20Extended(address(tokenB)).symbol()
+            )
+        );
+
+        return string(abi.encodePacked("NoHedgeUniV3StablesJoint(", ab, ")"));
+    }
+
+    /*
+     * @notice
+     *  Function returning the liquidity amount of the LP position
+     * @return liquidity from positionInfo
+     */
+    function balanceOfPool() public view override returns (uint256) {
+        IUniswapV3Pool.PositionInfo memory positionInfo = _positionInfo();
+        return positionInfo.liquidity;
+    }
+
+    /*
+     * @notice
+     *  Function available for vault managers to set the CRV pool to use for swaps
+     * @param newPool, new CRV pool address to use
+     */
+    function setCRVPool(address newPool) external onlyVaultManagers {
+        crvPool = newPool;
+    }
+
+    /*
+     * @notice
+     *  Function available for vault managers to set the boolean value deciding wether
+     * to use the uni v3 pool for swaps or a CRV pool
+     * @param newUseCRVPool, new boolean value to use
+     */
+    function setUseCRVPool(bool newUseCRVPool) external onlyVaultManagers {
+        useCRVPool = newUseCRVPool;
+    }
+
+    /*
+     * @notice
+     *  Function available for vault managers to set the number of ticks on each side of 
+     * current tick to provide liquidity to
+     * @param newTicksFromCurrent, new value to use
+     */
+    function setTicksFromCurrent(uint24 newTicksFromCurrent) external onlyVaultManagers {
+        ticksFromCurrent = newTicksFromCurrent;
+    }
+
+    /*
+     * @notice
+     *  Function available for vault managers to set min & max values of the position. If,
+     * for any reason the ticks are not the value they should be, we always have the option 
+     * to re-set them back to the necessary value
+     * @param _minTick, lower limit of position
+     * @param _maxTick, upper limit of position
+     */
+    function setTicksManually(int24 _minTick, int24 _maxTick) external onlyVaultManagers {
```

Comment:
I would protect this function in case a position is open (as it could mess up with position retrieval).

it would mean to revert if a position is open AND a force argument is not passed.

Reply:
This function is actually created for that specific case, to be able to re-connect the joint to its position in case the ticks where set to 0 when they needed to stay with their respective values. As we discussed earlier it may make sense to have it protected with `onlyGovernance`, what do you guys think?

Reply:
the rug potential is limited (governance could always figure out why it can't get funds back, which is the max damage it can be done with this function)... so I'm fine leaving it like this